constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/utils/widget/MotionLabel.java

```diff
@@ -0,0 +1,392 @@
+
+int k = 0;
+            for (int i = 0; i < N; i++) {
+                int attr = a.getIndex(i);
+                if (attr == R.styleable.MotionLabel_android_text) {
+                    setText(a.getText(attr));
+                } else if (attr == R.styleable.MotionLabel_android_shadowColor) {
+                    mShadowColor = a.getColor(attr, mShadowColor);
+                } else  if (attr == R.styleable.MotionLabel_android_shadowDx) {
+                    mShadowDx =a.getFloat(attr,mShadowDx);
+                }  else  if (attr == R.styleable.MotionLabel_android_shadowDy) {
+                    mShadowDy =a.getFloat(attr,mShadowDy);
+                }  else  if (attr == R.styleable.MotionLabel_android_shadowRadius) {
+                    mShadowRadius =a.getFloat(attr,mShadowRadius);
+                }  else  if (attr == R.styleable.MotionLabel_textFillColor) {
+                    mTextColor = a.getColor(attr,mTextColor);
+                }  else  if (attr == R.styleable.MotionLabel_android_textSize) {
+                    mTextSize = a.getDimensionPixelSize(attr, mTextSize);
+                }  else  if (attr == R.styleable.MotionLabel_android_textStyle) {
+                    mStyleIndex = a.getInt(attr, mStyleIndex);
+                }  else  if (attr == R.styleable.MotionLabel_android_typeface) {
+                    mTypefaceIndex = a.getInt(attr, mTypefaceIndex);
+                }  else  if (attr == R.styleable.MotionLabel_textOutlineColor) {
+                    mTextOutlineColor = a.getColor(attr,mTextOutlineColor);
+                }  else  if (attr == R.styleable.MotionLabel_textOutlineThickness) {
+                    mTextOutlineThickness = a.getDimensionPixelSize(attr, mTextOutlineThickness);
+                } else if (attr == R.styleable.MotionLabel_borderRound) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        setRound(a.getDimension(attr, 0));
+                    }
+                } else if (attr == R.styleable.MotionLabel_borderRoundPercent) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        setRoundPercent(a.getFloat(attr, 0));
+                    }
+                }
+            }
+            a.recycle();
+        }
+        updatePaint();
+    }
+
+   void  updatePaint () {
+       mPaint.setColor(mTextColor);
+       mPaint.setStrokeWidth(mTextOutlineThickness);
+       mPaint.setStyle(Paint.Style.STROKE);
+       mPaint.setTypeface(Typeface.SERIF);
+       mPaint.setTextSize(mTextSize);
+       mPaint.setAntiAlias(true);
+    }
+
+    public void setText(CharSequence text) {
+        mText = text;
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+    }
+
+    void buildShape() {
+        mPath.reset();
+
+        String str = "Hello World";
```

Comment:
hm -- are you sure you want to submit this class yet? Seems not quite ready? what's this buildShape() function for?..